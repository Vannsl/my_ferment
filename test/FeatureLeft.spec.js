import { shallowMount } from '@vue/test-utils'
import FeatureLeft from '@/components/FeatureLeft'

const factory = () => {
  return shallowMount(FeatureLeft, {
    propsData: {
      headline: 'Lorem Ipsum',
      content: 'Lorem Ipsum dolor sit'
    }
  })
}

describe('FeatureLeft', () => {
  it('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has the property headline', () => {
    const wrapper = factory()
    expect(wrapper.props('headline')).toBe('Lorem Ipsum')
  })

  it('has the property content', () => {
    const wrapper = factory()
    expect(wrapper.props('content')).toBe('Lorem Ipsum dolor sit')
  })
})
