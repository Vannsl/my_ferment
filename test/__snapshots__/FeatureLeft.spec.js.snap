// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureLeft renders properly 1`] = `
"<div class=\\"flex flex-wrap flex-col-reverse sm:flex-row\\"><div class=\\"w-full sm:w-1/2 p-6 mt-6\\"></div> <div class=\\"w-full sm:w-1/2 p-6 mt-6\\"><div class=\\"align-middle\\"><h3 class=\\"text-3xl text-gray-800 font-bold leading-none mb-3\\">Lorem Ipsum</h3> <p class=\\"text-gray-600 mb-8\\">
        Lorem Ipsum dolor sit
        <br> <br>Images from:
        <a href=\\"https://undraw.co/\\" class=\\"text-orange-500 underline\\">undraw.co</a></p></div></div></div>"
`;
